### PR TITLE
feat(test): use correct primary key for tpch & nexmark ddl

### PR DESCRIPTION
**Pull request by wzzzzd**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/4860

### CODE REVIEW

**Review by BugenZhao — APPROVED** [review]

**Comment by BugenZhao** [comment]
Looks like there're records with the same primary key in our test cases. 🤣

File: src/frontend/test_runner/tests/testdata/tpch.yaml [hunk 1]
```diff
@@ -242,90 +258,83 @@
...
-                            BatchExchange { order: [], dist: HashShard(part.p_partkey) }
-                              BatchProject { exprs: [part.p_partkey, part.p_mfgr] }
-                                BatchFilter { predicate: (part.p_size = 4:Int32) AND Like(part.p_type, '%TIN':Varchar) }
-                                  BatchScan { table: part, columns: [part.p_partkey, part.p_mfgr, part.p_type, part.p_size], distribution: SomeShard }
+                              BatchScan { table: partsupp, columns: [partsupp.ps_partkey, partsupp.ps_suppkey, partsupp.ps_supplycost], distribution: UpstreamHashShard(partsupp.ps_partkey, partsupp.ps_suppkey) }
+                            BatchProject { exprs: [part.p_partkey, part.p_mfgr] }
+                              BatchFilter { predicate: (part.p_size = 4:Int32) AND Like(part.p_type, '%TIN':Varchar) }
+                                BatchScan { table: part, columns: [part.p_partkey, part.p_mfgr, part.p_type, part.p_size], distribution: UpstreamHashShard(part.p_partkey) }
```
Comment (BugenZhao) [inline src/frontend/test_runner/tests/testdata/tpch.yaml:264]: Looks like we should also insert an exchange here? cc @st1page @xxchan

Comment (st1page) [reply src/frontend/test_runner/tests/testdata/tpch.yaml:264]: strange... I will fix it.

Comment (st1page) [reply src/frontend/test_runner/tests/testdata/tpch.yaml:264]: https://github.com/singularity-data/risingwave/pull/4904

**Comment by BugenZhao** [comment]
After #4902, the distributed batch is not tested. We should first manually test it.

**Comment by wzzzzd** [comment]
After #4905, the distributed mode is enabled explicitly in batch tpch test. I think we can merge this PR now.

**Comment by wzzzzd** [comment]
@mergifyio requeue

**Pull request merged**